### DISCUSSION TITLE
How does one do a quick  add to stage?   I dont want the popup to open

### DISCUSSION THREAD

**Discussion by staycreativedesign** [NeogitOrg/neogit#1076] [General]
I just want to quickly add a file to stage 

currently i do  NeogitStatus command and basically it opens the git status

**Comment by CKolkey** [comment]
Staging/Unstage are both bound to the status buffer, so there's not a "public" way to do it, but if you want, you could dive into the internal call:
https://github.com/NeogitOrg/neogit/blob/master/lua/neogit/lib/git/index.lua#L104-L106

So, something like `require('neogit.lib.git.index').add({ "<filepath>" })` would work. But, the core workflow of Neogit is via the status buffer, so.. I don't think I'd add direct commands for those actions

**Comment by staycreativedesign** [comment]
sadness... :(

**Comment by staycreativedesign** [comment]
https://github.com/NeogitOrg/neogit/blob/master/doc/neogit.txt#L365C1-L365C57
:)

**Reply by CKolkey** [reply]
Yea, still won't let you stage/unstage, since those are not via a popup, but they will let you pull and do all the other stuff

**Comment by star-szr** [comment]
This is super basic and will not show errors so definitely room for improvement, but I use this which just runs the git command, no Neogit involved.

```lua
vim.keymap.set("n", "<leader>gw", function()
  -- TODO: Add error checking/reporting, check vim.v.shell_error?
  vim.cmd "silent !git add %:p"
end, { desc = "Git stage current buffer" })
```

**Comment by windowsrefund** [comment]
This is possible (easily) via gitsigns.nvim

**Reply by CKolkey** [reply]
This would be my recommendation for interacting with hunks outside the status buffer. Neogit is, first and foremost, a git TUI built inside neovim. Bypassing the TUI is a non-goal for me, and others have done this very well already.

**Reply by windowsrefund** [reply]
I should have read more closely as it seems the OP was looking for a way to `git add` a file for the first time. Gitsigns can't do that so I'm loving the little keymap provided by @star-szr
